node_modules/col-activities/lib/notifications/daily.js

```diff
@@ -0,0 +1,562 @@
+ * @param  {Function}   callback            Standard callback function
+ * @param  {Object}     callback.err        An error object, if any
+ * @param  {Object}     callback.assets     Assets that were recently commented on
+ * @api private
+ */
+var getCourseAssetData = function(ctx, users, callback) {
+  // Get the assets which have comments in the last 24 hours (including their parents if they have any)
+  var yesterday = get24HoursAgoTimestamp();
+  var options = {
+    'where': {
+      'course_id': ctx.course.id
+    },
+    'include': [
+      // Ideally, we'd just get the IDs and look up the objects in the `users` array ourselves
+      // to reduce data serialization. Unfortunately, Sequelize doesn't like this very much,
+      // so we simple get the basic user profiles
+      {'model': DB.User, 'as': 'users', 'required': true, 'attributes': UserConstants.BASIC_USER_FIELDS},
+
+      // The comments
+      {'model': DB.Comment, 'required': true, 'where': {'created_at': {'gt': yesterday}}, 'include': [
+        {'model': DB.Comment, 'required': false, 'as': 'parent'}
+      ]}
+    ],
+    'limit': null,
+    'subQuery': false
+  };
+  DB.Asset.findAll(options).complete(function(err, assets) {
+    if (err) {
+      log.error({'err': err, 'course': ctx.course.id}, 'Unable to get the assets for a course');
+      return callback(err);
+    }
+
+    assets = _.map(assets, function(asset) {
+      asset = asset.toJSON();
+
+      // Index the asset owners by their ID for quick lookups
+      asset.users = _.indexBy(asset.users, 'id');
+
+      // Add the commenters their user object
+      asset.comments = _.map(asset.comments, function(comment) {
+        comment = comment.toJSON();
+
+        comment.user = users[comment.user_id];
+        if (comment.parent) {
+          comment.parent.user = users[comment.parent.user_id];
+        }
+
+        return comment;
+      });
+
+      return asset;
+    });
+
+    return callback(null, assets);
+  });
+};
+
+/**
+ * Get the assets with recent chat messages
+ *
+ * @param  {Context}    ctx                     Standard context containing the current user and the current course
+ * @param  {Object}     users                   The users in the course (keyed by their user id)
+ * @param  {Function}   callback                Standard callback function
+ * @param  {Object}     callback.err            An error object, if any
+ * @param  {Object}     callback.whiteboards    Whiteboards with recent chat messages
+ * @api private
+ */
+var getCourseWhiteboardData = function(ctx, users, callback) {
+  // Get all the whiteboards that have chat messages in the past 24 hours
+  var yesterday = get24HoursAgoTimestamp();
+  var queryOptions = {
+    'where': {
+      'course_id': ctx.course.id
+    },
+    'include': [
+      {
+        'model': DB.User
+      },
+      {
+        'model': DB.Chat,
+        'where': {
+          'created_at': {
+            'gt': yesterday
```

Comment:
There's the potential for a chat message to be included in multiple emails as an email will include everything between 8 the previous day and the moment at which this is run (which will be a bit after 8)